tests/MongoDB.Driver.Core.Tests/Core/Bindings/ChannelSourceHandleTests.cs

```diff
@@ -155,10 +156,8 @@ public void Disposing_of_last_handle_should_dispose_of_connectioSource()
 
     internal static class ChannelSourceHandleReflector
     {
-        public static ReferenceCounted<IChannelSource> _reference(this ChannelSourceHandle obj)
-        {
-            var fieldInfo = typeof(ChannelSourceHandle).GetField("_reference", BindingFlags.NonPublic | BindingFlags.Instance);
-            return (ReferenceCounted<IChannelSource>)fieldInfo.GetValue(obj);
-        }
+        // private fields
+        public static bool _disposed(this ChannelSourceHandle obj) => (bool)Reflector.GetFieldValue(obj, nameof(_disposed));
```

Comment:
Since I was adding the _disposed method went ahead and refactored to use the new Reflector class.